src/main/java/com/three_tech_solutions/slot_app/services/implementations/PriceServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.three_tech_solutions.slot_app.services.implementations;
+
+import com.three_tech_solutions.slot_app.controllers.requests.PriceUpdateRequest;
+import com.three_tech_solutions.slot_app.controllers.responses.PriceUpdateResponse;
+import com.three_tech_solutions.slot_app.data.mappers.PriceMapper;
+import com.three_tech_solutions.slot_app.data.models.Price;
+import com.three_tech_solutions.slot_app.data.repositories.PriceRepository;
+import com.three_tech_solutions.slot_app.services.interfaces.PriceService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.UUID;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@Service
+@AllArgsConstructor
+public class PriceServiceImpl implements PriceService {
+
+    private final PriceRepository priceRepository;
+    @Override
+    public PriceUpdateResponse updatePriceAmount(UUID priceId, PriceUpdateRequest request) {
+        Price price = priceRepository.findById(priceId)
+                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "Precio no encontrado"));
+
+        price.setAmount(request.getAmount());
+        priceRepository.save(price);
+
+        return PriceMapper.toPriceUpdateResponse(price);
+    }
```

Comment:
Esto se podria refactorizar de la siguiente manera


```suggestion
    public PriceUpdateResponse updatePriceAmount(UUID priceId, PriceUpdateRequest request) {
        return priceRepository.findById(priceId)
                .map(price -> {
                    price.setAmount(request.getAmount());
                    priceRepository.save(price);
                    return PriceMapper.toPriceUpdateResponse(price);
                })
                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "Precio no encontrado"));
    }
```

De esta manera utilizamos las funcionas lambdas (`.map` y el `.orElseThrow`) y nos evitamos tener que setear en una variable el precio que obtenemos